src/engine/strat_engine/blockdev.rs

```diff
@@ -76,30 +74,29 @@ pub fn find_all() -> EngineResult<BTreeMap<PoolUuid, BTreeMap<DevUuid, BlockDev>
                                format!("Could not open {}", devnode.display()))
             }));
 
-        let mut buf = [0u8; 4096];
+        let mut buf = [0u8; BDA_STATIC_HDR_SIZE as usize];
```

Comment:
If we do this then there are other spots where we'd end up casting the other way ☹️ . I think conceptually `u64` is better because we're dealing with on-disk values that are not variable across 32 and 64bit CPU archs. But yeah when creating arrays these casts are ugly and get tiresome.